[Khor Jun Wei] iP

KJunWei opened the pull request:
### DukePro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

Commit:
Add Gradle support

KJunWei committed:
Level-1

KJunWei committed:
Level-2

KJunWei committed:
Level-3

KJunWei committed:
Level-4

KJunWei committed:
Level-5

KJunWei committed:
Level-6

KJunWei committed:
A-Enums

KJunWei committed:
A-TextUiTesting

KJunWei committed:
add javadocs

KJunWei committed:
clear code, abstract code into Task.java, TaskManager.java, add more javadocs

KJunWei committed:
add methods to achieve Level-7

KJunWei committed:
change string formats to LocalDate, input.txt, achieve Level-8

KJunWei committed:
Merge branch 'branch-Level-8'

KJunWei committed:
fix LocalDate bug

KJunWei committed:
add new classes, apply more oop, achieve A-MoreOOP

KJunWei committed:
add packages, modify runtest.bat

KJunWei committed:
Merge branch 'nus-cs2103-AY2223S1:add-gradle-support' into add-gradle-support

KJunWei committed:
add Todo.java and DukeTest.java

KJunWei committed:
package app as jar file

KJunWei committed:
add JUnit tests

KJunWei committed:
add javadocs

KJunWei committed:
edit code to match coding standard

KJunWei committed:
add methods to allow find method

KJunWei committed:
Merge branch 'branch-A-CodingStandard'

KJunWei committed:
Merge branch 'branch-Level-9'

zbz-lvlv commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,365 @@
+        /**
+         * Represents the task type.
+         */
+        private final Task.Task_type task_type;
```
The style guide specified that we need to use camelCase for variable names.

zbz-lvlv commented on src/main/java/duke/exception/DukeException.java:
```diff
@@ -0,0 +1,60 @@
+     * @param message message to determine the error
+     */
+    public DukeException(String message) {
+        if (message.equals("todo")) {
```
Long chains of if-else isn't very friendly. Perhaps you can consider suing switch-case or use a HashMap?

zbz-lvlv commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,322 @@
+        }
+
+        /**
+         * Checks whether task contains certain String
```
Some of your JavaDocs are missing a full stop at the end of the first sentence.

zbz-lvlv commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,125 @@
+    /**
+     * Represents an indentation for replies.
+     */
+    private static final String INDENTATION = "     ";
```
I like how you've follow the convention for constants here. Good job :)

zbz-lvlv commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,81 @@
+
+    /**
+     * Represents a constructor method for Duke class
+     * @param filepath
```
Please give a description for the filepath parameter.

zbz-lvlv reviewed (commented):
Good job! Overall good conformance to the style guide. I like the way you designed the Command class as well, very nice application of OOP principles!

KJunWei committed:
add CheckStyle

KJunWei committed:
add step 1-3 of JavaFX

KJunWei committed:
edit code to match checkstyle

KJunWei committed:
use FXML

KJunWei committed:
add varargs for method in Ui.java

KJunWei committed:
add GUI

KJunWei committed:
Remove unnecessary code in Duke.java, fix bugs

Chustinjeng commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,372 @@
+    /**
+     * Represents the type of commands the class understands.
+     */
+    public enum ActionKeywords {
+        DEADLINE,
+        DELETE,
+        EVENT,
+        FIND,
+        LIST,
+        MARK,
+        TODO,
+        UNMARK,
+    }
```
good job of using enums!

Chustinjeng commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,372 @@
+    public static Command of(String s) throws DukeException {
+        String[] splitResponse = s.split(" ");
+        String keyword = splitResponse[0];
+        if (splitResponse.length == 1) {
+            if (keyword.equals("todo") || keyword.equals("deadline") || keyword.equals("event")
+                    || keyword.equals("delete") || keyword.equals("mark") || keyword.equals("unmark")
+                    || keyword.equals("find")) {
+                throw new DukeException(keyword);
+            } else if (keyword.equals("list")) {
+                return new ListCommand();
+            } else if (keyword.equals("bye")) {
+                return new ExitCommand();
+            } else {
+                throw new DukeException("unknown");
+            }
```
can perhaps use switch here rather than if else

Chustinjeng commented on src/main/java/duke/exception/DukeException.java:
```diff
@@ -0,0 +1,60 @@
+     * Constructor for DukeException.
+     * @param message message to determine the error
+     */
+    public DukeException(String message) {
```
can try to use switch instead :)

Chustinjeng reviewed (commented):
I like how you used enums in your code. Overall good flow and use of OOP principles. Great job! 👍 💯

KJunWei committed:
Add assertion statements to TaskManager.java

There are various methods in TaskManager.java that
takes in the location of the task as a parameter.

These methods should ensure the location is not a negative number.

Let's include assertion lines to ensure the number given will
not be negative.

KJunWei committed:
Resolve feedback by classmates

Some classmates have given constructive
constructive feedback via Github.

Apply what they recommended improves code quality
in terms of Checkstyle.

Let's remove long if-else blocks and
ensure javadoc comments are well-formatted.

KJunWei committed:
Add streams in TaskManager.java

KJunWei committed:
Merge pull request #1 from KJunWei/branch-A-Assertions

Add assertion statements to TaskManager.java

KJunWei committed:
Merge branch 'master' into branch-A-CodeQuality

KJunWei committed:
Merge branch 'master' into branch-A-Streams

KJunWei committed:
Merge pull request #2 from KJunWei/branch-A-CodeQuality

Branch a code quality

KJunWei committed:
Merge pull request #3 from KJunWei/branch-A-Streams

Branch a streams

KJunWei committed:
Add CI

KJunWei committed:
Add reminders command as an extension

KJunWei committed:
Address issues based on code quality feedback

KJunWei committed:
Clean code, add Ui.png

KJunWei committed:
Update README.md

KJunWei committed:
Update README.md

KJunWei committed:
Update README.md

KJunWei committed:
Update README.md

KJunWei committed:
Edit code to ensure shadowJar works

KJunWei committed:
Edit javadocs

KJunWei committed:
Edit minor method names

KJunWei committed:
Edit code to match coding standard

KJunWei committed:
Edit code to match coding standard
